handbook/engineering/search/index.md

```diff
@@ -2,9 +2,93 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+Our current focus is documented in the [tracking issue for the current milestone](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast, I can get to what I’m looking for quickly
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write.
+The answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can figure out how to find what they are looking for, the syntax is clear, new users understand what options they have for searching.
+- **Sharable:** Sourcegraph search creates network effects that provide compounding value with scale. 
+
 ## Goals
 
-Our current goals are documented in the [search team's current tracking issue](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+_Updated 2020-08-27_
+
+## Long term (~1 year)
+
+Sourcegraph search is far and away the best code search solution, search results are fast, and I can find what I’m looking for quickly.
+
+## Medium term (3-6 months)
+
+In the next 6 months we want to be halfway to the long term goal. To achieve this, we need to accomplish the following outcomes:
+
+- Getting to search results faster through a better search experience.
+- What do we need to do to get to the speed of grep.app?
+- Scale fast search to 500k repositories.
+
+## Short term (1-3 months)
+
+### Fast
+
+- **Streaming search**
+  - **Problem:** We want to make significant improvements to search that are blocked by not having the infrastructure needed.
+  - **Owner:** Keegan
+  - **Status:** In progress
+
+### Expressive
+
+- **Hierarchical search (AND/OR/NOT)**
```

Comment:
@rvantonder I think you had an idea for a different name for this, what did you call it?

Reply:
Search expressions. For the general audience, I wrote up why I'm adopting this terminology in https://github.com/sourcegraph/sourcegraph/issues/13126. "Search expressions" doesn't "have" to be the product label we slap onto this (we could simply go with "Sourcegraph's Query Language" or "Sourcegraph Queries" that's expanded over time. As far as internal goals and conveying the work, search expressions is more accurate than previous terms :-)

>

Reply:
I like _search expressions_ a lot. It feels intuitive to gather that it's about combining two or more search expressions with operators. 

If we go this route we should what a ‘search expression’ is. I think it’s easy to confuse it with phrasing that implies similar meaning in our docs today e.g. 'search patterns' used throughout the [docs](https://docs.sourcegraph.com/user/search) and there's also [regular expression search](https://docs.sourcegraph.com/user/search/queries#regular-expression-search).
There's related discussion underway in this [product terms RFC](https://docs.google.com/document/d/1SHky6nodPs1w_zRXz24jB2nq5LbMCJl1u7hcLTGHDL8/edit#heading=h.trqab8y0kufp). Search expressions in this RFC are defined as the smallest units of valid search queries (without operators AND/OR/NOT) and my understanding is the phrasing 'search pattern' should be replaced with 'search expression' in the docs.

Perhaps “Combining Search Expressions” is a clearer goal here as it describes the ability to compose individual search expressions including the ability to nest these searches.

Reply:
I like "Search expressions (AND/OR/NOT)" a lot. I agree that it feels more intuitive and I can easily tell that it's about combining multiple search expressions :)

If we go this route we should define and agree on what the definition of a "search expression" is. I think it’s easy to confuse it with phrasing that implies similar meaning in our docs today e.g. "search patterns" is used throughout the [docs](https://docs.sourcegraph.com/user/search) and there's also ["regular expression search"](https://docs.sourcegraph.com/user/search/queries#regular-expression-search).
There's related discussion underway in this [product terms RFC](https://docs.google.com/document/d/1SHky6nodPs1w_zRXz24jB2nq5LbMCJl1u7hcLTGHDL8/edit#heading=h.trqab8y0kufp). Search expressions in this RFC are defined as smallest units of valid search queries (without operators AND/OR/NOT) and my understanding is the phrasing "search pattern" should be replaced with "search expression" in the docs. What do you think?

Perhaps "Combining Search Expressions" is a clearer goal as it describes the ability to compose/combine the individual search expressions (following the definition of a search expression above) including the ability to nest these searches.